python/sdist/custom_commands.py

```diff
@@ -0,0 +1,223 @@
+"""Custom setuptools commands for AMICI installation"""
+
+import glob
+import os
+import sys
+import subprocess
+from shutil import copyfile
+
+from setuptools.command.build_ext import build_ext
+from setuptools.command.sdist import sdist
+from setuptools.command.install_lib import install_lib
+from setuptools.command.develop import develop
+from setuptools.command.install import install
+from setuptools.command.build_clib import build_clib
+
+from amici.setuptools import generateSwigInterfaceFiles
+
+
+class my_install(install):
+    """Custom install to handle extra arguments"""
+
+    # Passing --no-clibs allows to install the Python-only part of AMICI
+    user_options = install.user_options + [
+        ('no-clibs', None, "Don't build AMICI C++ extension"),
+    ]
+
+    def initialize_options(self):
+        install.initialize_options(self)
+        self.no_clibs = False
+
+    def finalize_options(self):
+        if self.no_clibs:
+            self.no_clibs = True
+        install.finalize_options(self)
+
+    def run(self):
+        install.run(self)
+
+
+def compile_parallel(self, sources, output_dir=None, macros=None,
+            include_dirs=None, debug=0, extra_preargs=None,
+            extra_postargs=None, depends=None):
+    """Parallelized version of distutils.ccompiler.compile"""
+    macros, objects, extra_postargs, pp_opts, build = \
+        self._setup_compile(output_dir, macros, include_dirs, sources,
+                            depends, extra_postargs)
+    cc_args = self._get_cc_args(pp_opts, debug, extra_preargs)
+
+    # parallel compilation
+    num_threads = os.cpu_count()
```

Comment:
can we please have an option to limit the used cpus (env variable?).

Reply:
If we assume it may cause trouble, maybe we default to sequential and add `AMICI_PARALLEL_COMPILE=num_threads`